drivers/media/i2c/mira016.inl

```diff
@@ -0,0 +1,6501 @@
+	{
+		/* 400x400 10bit 100fps mode */
+		.width = 400,
+		.height = 400,
+		.crop = {
+			.left = MIRA016_PIXEL_ARRAY_LEFT,
+			.top = MIRA016_PIXEL_ARRAY_TOP,
+			.width = 400,
+			.height = 400},
+		.reg_list_pre_soft_reset = {
+			.num_of_regs = ARRAY_SIZE(full_400_400_100fps_10b_1lane_reg_pre_soft_reset),
+			.regs = full_400_400_100fps_10b_1lane_reg_pre_soft_reset,
+		},
+		.reg_list_post_soft_reset = {
+			.num_of_regs = ARRAY_SIZE(full_400_400_100fps_10b_1lane_reg_post_soft_reset),
+			.regs = full_400_400_100fps_10b_1lane_reg_post_soft_reset,
+		},
+		.min_vblank = MIRA016_MIN_VBLANK_360,
+		.max_vblank = MIRA016_MAX_VBLANK,
+		.hblank = MIRA016_HBLANK, // TODO
+		.bit_depth = 10,
+		.code = MEDIA_BUS_FMT_SGRBG10_1X10,
+	},
+	{
+		/* 400x400 12 bit 100fps mode */
+		.width = 400,
+		.height = 400,
+		.crop = {.left = MIRA016_PIXEL_ARRAY_LEFT, .top = MIRA016_PIXEL_ARRAY_TOP, .width = 400, .height = 400},
+		.reg_list_pre_soft_reset = {
+			.num_of_regs = ARRAY_SIZE(full_400_400_100fps_12b_1lane_reg_pre_soft_reset),
+			.regs = full_400_400_100fps_12b_1lane_reg_pre_soft_reset,
+		},
+		.reg_list_post_soft_reset = {
+			.num_of_regs = ARRAY_SIZE(full_400_400_100fps_12b_1lane_reg_post_soft_reset),
+			.regs = full_400_400_100fps_12b_1lane_reg_post_soft_reset,
+		},
+		.min_vblank = MIRA016_MIN_VBLANK_200,
+		.max_vblank = MIRA016_MAX_VBLANK,
+		.hblank = MIRA016_HBLANK, // TODO
+		.bit_depth = 12,
+		.code = MEDIA_BUS_FMT_SGRBG12_1X12,
+	},
+	{
+		/* 400x400 8 bit 100fps mode */
+		.width = 400,
+		.height = 400,
+		.crop = {.left = MIRA016_PIXEL_ARRAY_LEFT, .top = MIRA016_PIXEL_ARRAY_TOP, .width = 400, .height = 400},
+		.reg_list_pre_soft_reset = {
+			.num_of_regs = ARRAY_SIZE(full_400_400_100fps_8b_1lane_reg_pre_soft_reset),
+			.regs = full_400_400_100fps_8b_1lane_reg_pre_soft_reset,
+		},
+		.reg_list_post_soft_reset = {
+			.num_of_regs = ARRAY_SIZE(full_400_400_100fps_8b_1lane_reg_post_soft_reset),
+			.regs = full_400_400_100fps_8b_1lane_reg_post_soft_reset,
+		},
+		.min_vblank = MIRA016_MIN_VBLANK_360,
+		.max_vblank = MIRA016_MAX_VBLANK,
+		.hblank = MIRA016_HBLANK, // TODO
+		.bit_depth = 8,
+		.code = MEDIA_BUS_FMT_SGRBG8_1X8,
+	},
+};
+
+struct mira016
+{
+	struct v4l2_subdev sd;
+	struct media_pad pad[NUM_PADS];
+
+	struct v4l2_mbus_framefmt fmt;
+
+	struct clk *xclk; /* system clock to MIRA016 */
+	u32 xclk_freq;
+
+	// struct gpio_desc *reset_gpio;
+	struct regulator_bulk_data supplies[MIRA016_NUM_SUPPLIES];
+
+	struct v4l2_ctrl_handler ctrl_handler;
+	struct v4l2_ctrl *pixel_rate;
+	struct v4l2_ctrl *vflip;
+	struct v4l2_ctrl *hflip;
+	struct v4l2_ctrl *vblank;
+	struct v4l2_ctrl *hblank;
+	struct v4l2_ctrl *exposure;
+	struct v4l2_ctrl *gain;
+	// custom v4l2 control
+	struct v4l2_ctrl *mira016_reg_w;
+	struct v4l2_ctrl *mira016_reg_r;
+	u16 mira016_reg_w_cached_addr;
+	u8 mira016_reg_w_cached_flag;
+
+	/* Current mode */
+	const struct mira016_mode *mode;
+	/* current bit depth, may defer from mode->bit_depth */
```

Comment:
Typo for differ.

And I see no route in the code where it can differ. Can you point it out to me?